### PR TITLE
feat: add dropdown menu

### LINKED ISSUE #15: Landing Page
## Description

- Create a landing page

---

## Reference

- Content to display on the landing page: https://www.nepalstock.com/
- Header: https://flowbite.com/docs/components/navbar/#navbar-with-search
- Footer: https://flowbite.com/docs/components/footer/#footer-with-logo

### PR DESCRIPTION
**Pull request by sumanjha108155**
## Description

This PR adds a dropdown menu to the features nav-item.

---

## Issue

- [x] Partially fixes - #15

### CODE REVIEW

File: src/components/Navbar.tsx [hunk 1]
```diff
@@ -9,17 +9,85 @@ const Navbar: React.FC = () => {
     setIsMenuOpen(!isMenuOpen);
   };
 
+  const [isDropdownOpen, setIsDropdownOpen] = useState(false);
```
Comment (nirajgiriXD) [inline src/components/Navbar.tsx:12]:
```suggestion
```
All the state declaration needs to be done at top.

File: src/components/Navbar.tsx [hunk 2]
```diff
@@ -9,17 +9,85 @@ const Navbar: React.FC = () => {
...
+  const dropdownItems = [
+    { label: "IPO Checker", href: "#" },
+    { label: "IPO/FPO Updates", href: "#" },
+    { label: "Trading Signal", href: "#" },
+    { label: "Portfolio Checker", href: "#" },
+    { label: "Charts", href: "#" },
+    { label: "Company Analyzer", href: "#" },
+    { label: "Compare Company", href: "#" },
+    { label: "Share Calculator", href: "#" },
+  ];
```
Comment (nirajgiriXD) [inline src/components/Navbar.tsx:47]: Keep all the constants in a separate file and name it as `constant.ts`. Keep the file in the same folder.

File: src/components/Navbar.tsx [hunk 3]
```diff
@@ -9,17 +9,85 @@ const Navbar: React.FC = () => {
...
+    <div className="cursor-pointer inline-block" onMouseOver={onMouseOver}>
+      <svg
+        className="fill-current h-4 w-4 ml-1"
+        xmlns="http://www.w3.org/2000/svg"
+        viewBox="0 0 20 20"
+      >
+        <path d="M9.293 12.95l.707.707L15.657 8l-1.414-1.414L10 10.828 5.757 6.586 4.343 8z" />
+      </svg>
+    </div>
+  );
```
Comment (nirajgiriXD) [inline src/components/Navbar.tsx:59]: This needs to be a separate component.

File: src/components/Navbar.tsx [hunk 4]
```diff
@@ -9,17 +9,85 @@ const Navbar: React.FC = () => {
...
+                className="block text-gray-800 hover:text-blue-500 "
+              >
+                {subItem.label}
+              </a>
+            </div>
+          ))}
+        </div>
+      </>
+    );
+  };
```
Comment (nirajgiriXD) [inline src/components/Navbar.tsx:85]: This also needs to be a component.

File: src/components/Navbar.tsx [hunk 5]
```diff
@@ -9,17 +9,85 @@ const Navbar: React.FC = () => {
...
+    setIsDropdownOpen(true);
+  };
+
+  const handleDropdownMouseOut = () => {
+    // Close the dropdown when the mouse leaves the dropdown
+    setIsDropdownOpen(false);
+  };
+
   const navLinks = [
     { label: "Home", href: "#" },
```
Comment (nirajgiriXD) [inline src/components/Navbar.tsx:32]: We will have to remove the `home` navitem as we already have a logo that acts as `home` button, and save us precious space on the navbar for another navitem.

**Review by nirajgiriXD — CHANGES_REQUESTED** [review]

File: src/components/Navbar.tsx (hunk 5 above)
Comment (nirajgiriXD) [inline src/components/Navbar.tsx:32]: We will have to remove the `home` navitem as we already have a logo that acts as `home` button, and save us precious space on the navbar for another navitem.

File: src/components/Navbar.tsx (hunk 2 above)
Comment (sumanjha108155) [reply src/components/Navbar.tsx:46]: ok will do that

File: src/components/Navbar.tsx (hunk 3 above)
Comment (sumanjha108155) [reply src/components/Navbar.tsx:58]: ok will do that

File: src/components/Navbar.tsx (hunk 4 above)
Comment (sumanjha108155) [reply src/components/Navbar.tsx:84]: ok will do that

File: src/components/Navbar.tsx (hunk 5 above)
Comment (sumanjha108155) [reply src/components/Navbar.tsx:31]: ok will do that

**Review by nirajgiriXD — APPROVED** [review]

**Pull request merged**